e2e/discoverSheetFlow.spec.js

```diff
@@ -0,0 +1,199 @@
+      // Set the pin
+      await Helpers.authenticatePin('1234');
+      // Confirm it
+      await Helpers.authenticatePin('1234');
+    }
+    await Helpers.delay(1000);
+    await Helpers.checkIfVisible('wallet-screen');
+  });
+
+  it('Should navigate to Discover screen after tapping Discover Button', async () => {
+    await Helpers.delay(2000);
+    await Helpers.tap('discover-button');
+    await Helpers.delay(3000);
+    await Helpers.checkIfVisible('top-movers-section');
+    await Helpers.checkIfVisible('dpi-button');
+    await Helpers.checkIfVisible('lists-section');
+    await Helpers.checkIfVisible('pools-section');
+  });
+
+  it('Should show the camera if Discover is minimized', async () => {
+    await Helpers.delay(1000);
+    await Helpers.swipe('discover-header', 'down');
+    await Helpers.delay(1000);
+    await Helpers.checkIfVisible('scanner-header');
+    await Helpers.checkIfNotVisible('pools-section');
+  });
+
+  it('Should open Discover Search on pressing search fab', async () => {
+    await Helpers.delay(1000);
+    await Helpers.tap('search-fab');
+    await Helpers.swipe('discover-header', 'up');
+    await Helpers.delay(2000);
+    await Helpers.checkIfVisible('done-button');
+  });
+
+  it('Should search and open expanded state for SOCKS', async () => {
+    await Helpers.delay(1000);
+    await Helpers.typeText('discover-search-input', 'SOCKS\n', true);
+    await Helpers.delay(8000);
+    await Helpers.checkIfVisible(
+      'discover-currency-select-list-exchange-coin-row-SOCKS'
+    );
+    await Helpers.checkIfNotVisible(
+      'discover-currency-select-list-exchange-coin-row-ETH'
+    );
+    await Helpers.tap('discover-currency-select-list-exchange-coin-row-SOCKS');
+    await Helpers.delay(2500);
+    await Helpers.checkIfVisible('chart-header-Unisocks');
+  });
+
+  it('Should add Unisocks to Watchlist & remove from Favorites', async () => {
+    await Helpers.tap('add-to-list-button');
+    await Helpers.delay(2000);
+    await Helpers.checkIfVisible('add-token-sheet');
+    await Helpers.tap('add-to-watchlist');
+    await Helpers.delay(1000);
+    await Helpers.tap('remove-from-favorites');
+    await Helpers.delay(1000);
+    await Helpers.tap('close-action-button');
+  });
+
+  it('Should close expanded state and return to search', async () => {
+    await Helpers.delay(4000);
+    await Helpers.swipe('expanded-state-header', 'down');
+    await Helpers.delay(1000);
+    await Helpers.checkIfNotVisible(
+      'discover-currency-select-list-exchange-coin-row-ETH'
+    );
+  });
+
+  it('Should close search and return to Discover Home on pressing Done', async () => {
+    await Helpers.tap('done-button');
+    await Helpers.delay(1000);
+    await Helpers.checkIfVisible('top-movers-section');
+    await Helpers.checkIfVisible('dpi-button');
+    await Helpers.checkIfVisible('lists-section');
+    await Helpers.checkIfVisible('pools-section');
```

Comment:
This is a bit fragile since we will be adding more stuff soon, so what's visible or not will be likely changing often.
How about having a testID on the view that wraps those 3 sections?